index.js

```diff
@@ -1,4 +1,5 @@
 'use strict';
 const pReflect = require('p-reflect');
+const pMap = require('p-map');
 
-module.exports = iterable => Promise.all(iterable.map(pReflect));
+module.exports = (iterable, opts) => pMap(iterable.map(pReflect), i => i, opts).then(prms => Promise.all(prms));
```

Comment:
This is not correct. You need to use the mapper of `pMap`, not array map. That's how it controls concurrency. Also use more readable variable names.

Reply:
Unfortunately that is not possible since `pMap` calls `Promise.resolve` and maps the value on a `then` function. This means that is we have a failing promise on `pSettle`, none will get resolved. Hence me wrapping the array values with `pReflect` first.
All this now leads me to believe i need to implement concurrency by hand... instead of using `pMap` :/

Reply:
Seems so. Shouldn't be too hard to implement manually if you use https://github.com/sindresorhus/p-limit

Reply:
Will do it as soon as i can.